sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/datasource/notification/create/FetchSubscriberDbConnector.groovy

```diff
@@ -0,0 +1,103 @@
+package life.qbic.samplenotificator.datasource.notification.create
+
+import groovy.util.logging.Log4j2
+import life.qbic.business.exception.DatabaseQueryException
+import life.qbic.business.subscription.fetch.FetchSubscriberDataSource
+import life.qbic.business.subscription.Subscriber
+import life.qbic.datamodel.samples.Status
+import life.qbic.samplenotificator.datasource.database.ConnectionProvider
+import life.qbic.samplenotificator.datasource.database.DatabaseSession
+
+import java.sql.Connection
+import java.sql.PreparedStatement
+import java.sql.Timestamp
+import java.time.Instant
+import java.time.LocalDate
+import java.time.ZoneId
+
+/**
+ * <h1>A database connector to retrieve information for sending notifications</h1>
+ *
+ * <p>This connector reads the notification queue table and finds the associated people that subscribed for a notification</p>
+ *
+ * @since 1.0.0
+ *
+*/
+@Log4j2
+class FetchSubscriberDbConnector implements FetchSubscriberDataSource{
+    private ConnectionProvider connectionProvider
+
+    FetchSubscriberDbConnector(ConnectionProvider connectionProvider){
+        this.connectionProvider = connectionProvider
+    }
+
+    @Override
+    Map<String, Status> getNotificationsForDay(LocalDate day){
+        Instant startOfDay = day.atStartOfDay().atZone(ZoneId.systemDefault()).toInstant()
+        Instant endOfDay = day.plusDays(1).atStartOfDay().minusNanos(1).atZone(ZoneId.systemDefault()).toInstant()
+        String sqlQuery = SELECT_NOTIFICATIONS + " WHERE arrival_time BETWEEN ? AND ?"
+        Map<String, Status> foundNotifications = new HashMap<>()
+
+        try{
+            Connection connection = connectionProvider.connect()
+
+            connection.withCloseable {
+                PreparedStatement preparedStatement = it.prepareStatement(sqlQuery)
+                preparedStatement.setTimestamp(1, Timestamp.from(startOfDay))
+                preparedStatement.setTimestamp(2, Timestamp.from(endOfDay))
+                preparedStatement.execute()
+
+                def resultSet = preparedStatement.getResultSet()
+                while (resultSet.next()) {
+                    String sampleCode = resultSet.getString("sample_code")
+                    Status status = Status.valueOf(resultSet.getString("sample_status"))
+                    foundNotifications.put(sampleCode,status)
+                }
+            }
+        }catch(Exception exception){
+            throw new DatabaseQueryException(exception.message)
+        }
+        return foundNotifications
+    }
+
+    @Override
```

Comment:
This is not necessary since JavaDoc is automatically copied if not provided for overwritten methods
https://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#inheritingcomments